draft-ietf-ppm-dap.md

```diff
@@ -1377,20 +1389,30 @@ This message consists of:
 * `prepare_inits`: the sequence of `PrepareInit` messages constructed in the
   previous step.
 
-Finally, the Leader sends a PUT request to
-`{helper}/tasks/{task-id}/aggregation_jobs/{aggregation-job-id}`. The payload
-is set to `AggregationJobInitReq` and the media type is set to
-"application/dap-aggregation-job-init-req".
+Finally, the Leader sends an HTTP PUT request to
+`{helper}/tasks/{task-id}/aggregation_jobs/{aggregation-job-id}` with a media
+type of "application/dap-aggregation-job-init-req" and a body containing the
+`AggregationJobInitReq`.
 
 The Leader MUST authenticate its requests to the Helper using a scheme that
 meets the requirements in {{request-authentication}}.
 
-The Helper's response will be an `AggregationJobResp` message (see
-{{aggregation-helper-init}}. The response's `prepare_resps` must include exactly
-the same report IDs in the same order as the Leader's `AggregationJobInitReq`.
-Otherwise, the Leader MUST abort the aggregation job.
-
-[[OPEN ISSUE: consider relaxing this ordering constraint. See issue#217.]]
+The Helper responds with HTTP status 201 Created. If the response has a media
+type of "application/dap-aggregation-job-resp" with a body containing an
+`AggregationJobResp`, the Leader proceeds onward (see
+{{aggregation-helper-init}}). Otherwise, the Leader polls the aggregation job
+by sending HTTP GET requests to
+`{helper}/tasks/{task-id}/aggregation_jobs/{aggregation-job-id}?step=0`. If the
```

Comment:
Where did `?step=0` come from?

Reply:
"step" refers to the step of the aggregation job:

* step 0 is the initial `AggregationJobInitReq`
* step 1 is the first `AggregationJobContinueReq`
* step 2 is the second `AggregationJobContinueReq`
* and so on.

Note that, for Prio3, aggregation is complete after step 0.

We decided to add the `step` parameter to the GET request so that the helper knows how to respond if the leader somehow ends up in a weird state.